oc_cdtapi/OnecAPI.py

```
@@ -0,0 +1,229 @@
+            logging.debug("Generated query string: %s", query_string)
+            
+            response = self.get("clients", params={"query": query_string})
+            
+            if response.status_code == 200:
+                return self._parse_clients_response(response)
+            else:
+                error_msg = (
+                    f"Client request failed with status {response.status_code}. "
+                    f"Response: {response.json()}"
+                )
+                logging.warning(error_msg)
```

Ruff (0.15.4)

[warning] `warning()` call on root logger

(LOG015)